modules/homefinder/widgets/class-affiliate-search-widget.php

```diff
@@ -0,0 +1,208 @@
+<?php
+/**
+ * HomeFinder Affiliate Search Widget (http://www.homefinder.com/widgets/)
+ *
+ * @package RE-PRO
+ */
+
+	/* Exit if accessed directly. */
+if ( ! defined( 'ABSPATH' ) ) { exit; }
+
+/**
+ * HomeFinderAffiliateSearch class.
+ *
+ * @extends WP_Widget
+ */
+class HomeFinderAffiliateSearch extends WP_Widget {
+
+	/**
+	 * __construct function.
+	 *
+	 * @access public
+	 * @return void
+	 */
+	public function __construct() {
+
+		parent::__construct(
+			'homefinder_affiliate_search',
+			__( 'HomeFinder Affiliate Search', 're-pro' ),
+			array(
+				'description' => __( 'Display an affiliate search box from HomeFinder.com', 're-pro' ),
+				'classname'   => 're-pro re-pro-widget homefinder-widget homefinder-affiliate-search',
+				'customize_selective_refresh' => true,
+			)
+		);
+	}
+
+	/**
+	 * Widget function.
+	 *
+	 * @access public
+	 * @param mixed $args Arguments.
+	 * @param mixed $instance Instance.
+	 */
+	public function widget( $args, $instance ) {
+
+		$title = ! empty( $instance['title'] ) ? $instance['title'] : '';
+		$affiliate = ! empty( $instance['affiliate'] ) ? $instance['affiliate'] : '';
+		$primary_city = ! empty( $instance['primary_city'] ) ? $instance['primary_city'] : '';
+		$primary_state = ! empty( $instance['primary_state'] ) ? $instance['primary_state'] : '';
+		$nearby_city_1 = ! empty( $instance['nearby_city_1'] ) ? $instance['nearby_city_1'] : '';
+		$nearby_state_1 = ! empty( $instance['nearby_state_1'] ) ? $instance['nearby_state_1'] : '';
+		$nearby_city_2 = ! empty( $instance['nearby_city_2'] ) ? $instance['nearby_city_2'] : '';
+		$nearby_state_2 = ! empty( $instance['nearby_state_2'] ) ? $instance['nearby_state_2'] : '';
+		$nearby_city_3 = ! empty( $instance['nearby_city_3'] ) ? $instance['nearby_city_3'] : '';
+		$nearby_state_3 = ! empty( $instance['nearby_state_3'] ) ? $instance['nearby_state_3'] : '';
+		$nearby_city_4 = ! empty( $instance['nearby_city_4'] ) ? $instance['nearby_city_4'] : '';
+		$nearby_state_4 = ! empty( $instance['nearby_state_4'] ) ? $instance['nearby_state_4'] : '';
+
+		echo $args['before_widget'];
+
+		echo $args['before_title'] . esc_attr( $title ) . $args['after_title'];
+
+		$homefinder_widgets = new HomeFinderWidgets();
+
+		$homefinder_widgets->get_affiliates_widget( 'search', $instance );
+
+		echo $args['after_widget'];
+	}
+
+	/**
+	 * Form function.
+	 *
+	 * @access public
+	 * @param mixed $instance Instance.
+	 * @return void
+	 */
+	public function form( $instance ) {
```

Comment:
__[phpmd]__
* The method form() has a Cyclomatic Complexity of 13. The configured cyclomatic complexity threshold is 10.

<!-- Comment by [SideCI](https://sideci.com) -->